source/opengl/opengl_effect_compiler.cpp

```diff
@@ -2226,7 +2226,7 @@ namespace reshade::opengl
 
 		auto &uniform_storage = _runtime->get_uniform_value_storage();
 
-		if (_uniform_storage_offset + _uniform_buffer_size >= uniform_storage.size())
+		if (_uniform_storage_offset + _uniform_buffer_size >= static_cast<ptrdiff_t>(uniform_storage.size()))
```

Comment:
I don't see the reasoning behind this. `ptrdiff_t` is signed, `size_t` is not and the overall size can't be signed either, so this only crates problems but doesn't solve any?
EDIT: I see this was done to not run into a warning due to signed/unsigned mismatch on the comparison. It would problably make more sense to make `_uniform_storage_offset` and `_uniform_storage_size` of type `size_t` instead? Because they are used in various `size_t` expressions already anyway.